### PR TITLE
[docs] Step-by-step how to migrate existing users to SSO

### PR DESCRIPTION
**Pull request by keith-kurak**
# Why
Due to corporate policies, often, once an organization is using SSO, it's not really practical for them to not transition everyone over to SSO, even if we recommend they don't. We get a lot of questions about how you make this transition, so let's spell it out how it could work. By outlining what you need to do, this should further reinforce the answer to questions about what is not currently possible (an automated migration).

# How
- Rework text about mixed SSO and non-SSO users to acknowledge the transition state
- Provide a step by step that individual users plus account administrators could follow to get transitioned over

NOTE: this mentions a visual indication of an SSO user that is in progress (https://github.com/expo/universe/pull/13867).

### CODE REVIEW

File: docs/pages/accounts/sso.mdx [hunk 1]
```diff
@@ -51,9 +51,19 @@ SSO users are like regular users. However, there are a few known exceptions:
 
 ## SSO administration
 
-Both new organizations and existing organizations can enable SSO as a sign in option. Organizations with existing non-SSO members can enable SSO, and both non-SSO users and SSO users can continue to be added as members of SSO organizations.
+Both new organizations and existing organizations can enable SSO as a sign in option. Organizations with existing non-SSO members can enable SSO and then direct new members to the SSO sign-in page, while existing users continue to use their current Expo credentials. To support external contributors, SSO-enabled organizations also support inviting additional non-SSO users via email.
+
+### Transitioning existing users to SSO
+Compared to regular users, SSO users have an exclusive relationship with their organization. Thus, existing users cannot be directly migrated to SSO. However, a user that's already a member of your organization can create a second account by going to the [SSO login page](https://expo.dev/sso-login). Then, their old user can be removed from the organization.
```
Comment (wschurman) [inline docs/pages/accounts/sso.mdx:57]: maybe, not sure about this change, up to you.
```suggestion
Compared to regular users, SSO users have an exclusive relationship with their organization (to maintain security requirements of some organizations). Thus, existing users cannot be directly migrated to SSO. However, a user that's already a member of your organization can create a second account by going to the [SSO login page](https://expo.dev/sso-login). Then, their old user can be removed from the organization.
```

File: docs/pages/accounts/sso.mdx [hunk 2]
```diff
@@ -51,9 +51,19 @@ SSO users are like regular users. However, there are a few known exceptions:
...
+
+To transition from using a regular Expo account to an SSO account, follow these steps:
+1. Go to the [SSO login page](https://expo.dev/sso-login) and follow the prompts, entering your organization name, creating a new Expo username, and logging in to your identity provider.
+2. By default, your new SSO user will have the View Only role. If you need a different role, ask an Admin or Owner to update your role in [member settings](https://expo.dev/accounts/[account]/settings/members).
+3. At this time, the Admin or Owner can remove your old user from the organization. In [member settings](https://expo.dev/accounts/[account]/settings/members), the list of organization members indicates whether a user is an SSO or non-SSO user. The Admin or Owner can click the dropdown next to the old user and click **Remove member**.
+4. If you no longer need your old user account, log out of your new SSO account, then login to your old account and go to [user settings](https://expo.dev/settings). Scroll down and click **Delete Account**. **Note that this will delete any projects under  your old user account.** It will not affect any projects owned by the organization.
```
Comment (wschurman) [inline docs/pages/accounts/sso.mdx:63]:
```suggestion
4. If you no longer need your old user account, log out of your new SSO account, then login to your old account and go to [user settings](https://expo.dev/settings). Scroll down and click **Delete Account**. **Note that this will delete any projects under your old user account.** It will not affect any projects owned by the organization.
```

File: docs/pages/accounts/sso.mdx [hunk 3]
```diff
@@ -51,9 +51,19 @@ SSO users are like regular users. However, there are a few known exceptions:
...
+
+> If you wish to reuse your old username on your new SSO user account, you can go to [user settings](https://expo.dev/settings) under your old user and rename it prior to creating your SSO account. You can also rename your SSO user account's Expo username later.
```
Comment (wschurman) [inline docs/pages/accounts/sso.mdx:65]:
```suggestion
> If you wish to reuse your old username on your new SSO user account, you can go to [user settings](https://expo.dev/settings) under your old user and rename it prior to creating your SSO account. Alternatively, you can rename your SSO user account's Expo username after deleting your old user.
```

**Review by wschurman — APPROVED** [review]
Couple optional comments but nothing blocking.

File: docs/pages/accounts/sso.mdx (hunk 3 above)
Comment (lzkb) [reply docs/pages/accounts/sso.mdx:65]: There was a comment from someone from Brex that they had to delete their existing user that used their work email before they could create their SSO user that uses their work email. I'm wondering if we need to explicitly say that you don't need to delete your existing account before creating an SSO user.

File: docs/pages/accounts/sso.mdx (hunk 1 above)
Comment (keith-kurak) [reply docs/pages/accounts/sso.mdx]: I like the idea, not sure about the wording - I just pushed something a little different here. My goal with the paragraph is to echo the restrictions / differences here: https://docs.expo.dev/accounts/sso/#sso-user-restrictions as a reason why the accounts don't lend themselves well to migration. IMO the difference transcends any particular organization's security requirements; org-level ownership of a user just isn't really compatible with the possibility of multi-org access. (I'm thinking of the user who might say, "well, those aren't _my_ requirements, give me the migration!)

File: docs/pages/accounts/sso.mdx (hunk 3 above)
Comment (keith-kurak) [reply docs/pages/accounts/sso.mdx]: @lzkb I was thinking about this, too, and wasn't sure where to slot it in without getting too wordy. I just tried adding it to the callout at the bottom of the "transition" section. Feedback most welcome!

**Comment by keith-kurak** [comment]
Also added a note about needing to keep a single non-SSO member, since I think that would come up in discussions where 100% SSO adoption is desired. Since it very much relates to what would happen if SSO were disabled, I put it in that section.

File: docs/pages/accounts/sso.mdx [hunk 4]
```diff
@@ -51,9 +51,18 @@ SSO users are like regular users. However, there are a few known exceptions:
...
 
-People with existing non-SSO accounts can evaluate SSO by signing in with SSO, thus creating a second account, but we don't recommend this for all members. For simplicity, we recommend that new organization members use SSO to sign up but existing members continue to sign in as they normally would with their credentials. Migration of a non-SSO user to a SSO user is not currently supported.
+### Transitioning existing users to SSO
+Regular user accounts are owned by the individual user and joined to organizations as needed, while SSO users belong exclusively to their organization. Thus, existing users cannot be directly migrated to SSO. However, a user that's already a member of your organization can create a second account by going to the [SSO login page](https://expo.dev/sso-login). Then, their old user can be removed from the organization.
```
Comment (wschurman) [inline docs/pages/accounts/sso.mdx:57]:
```suggestion
Regular users may be a member of one or many personal, team, and organization accounts while SSO users belong exclusively to their organization account. Thus, existing users cannot be directly converted into SSO users. However, a regular user that's already a member of your organization may create a second user by going to the [SSO login page](https://expo.dev/sso-login). Then, their regular user can be removed from the organization.
```

File: docs/pages/accounts/sso.mdx [hunk 5]
```diff
@@ -51,9 +51,18 @@ SSO users are like regular users. However, there are a few known exceptions:
...
+
+To transition from using a regular Expo account to an SSO account, follow these steps:
+1. Go to the [SSO login page](https://expo.dev/sso-login) and follow the prompts, entering your organization name, creating a new Expo username, and logging in to your identity provider.
+2. By default, your new SSO user will have the View Only role. If you need a different role, ask an Admin or Owner to update your role in [member settings](https://expo.dev/accounts/[account]/settings/members).
+3. At this time, the Admin or Owner can remove your old user from the organization. In [member settings](https://expo.dev/accounts/[account]/settings/members), the list of organization members indicates whether a user is an SSO or non-SSO user. The Admin or Owner can click the dropdown next to the old user and click **Remove member**.
```
Comment (wschurman) [inline docs/pages/accounts/sso.mdx:62]: That regular user can also leave the account themselves, but maybe that adds more complication to this process.

File: docs/pages/accounts/sso.mdx [hunk 6]
```diff
@@ -69,6 +78,8 @@ If you wish to remove them ahead of that time or you wish to remove them to clea
 
 An active Enterprise Plan is required to continue using SSO. [Contact us](https://expo.dev/contact) if you wish to discontinue the use of SSO or change your plan.
 
+In order to ensure uninterrupted access to your organization whether or not SSO is enabled, SSO organizations must keep at least one non-SSO user with the Owner role as a member.
```
Comment (wschurman) [inline docs/pages/accounts/sso.mdx:81]: I don't think we've ever verified this. In theory, I think a SSO org could replace all their regular users with SSO users. It just would theoretically add risk of losing access if the IdP disabled all users.

File: docs/pages/accounts/sso.mdx (hunk 5 above)
Comment (keith-kurak) [reply docs/pages/accounts/sso.mdx:62]: Yeah, or the regular user can delete their account to close this loop right away. I wanted to focus as much as I could on things the account administrator could control, as they're the one accountable for ensuring the migration is complete.

File: docs/pages/accounts/sso.mdx (hunk 6 above)
Comment (keith-kurak) [reply docs/pages/accounts/sso.mdx:81]: Yeah, I doubt it breaks anything as long as SSO access is maintained, and I'm not aware of us enforcing this (though maybe we should?). I just want to set the expectation that they should keep this one account even after they're otherwise using SSO 100%, because there's no clear path to recovery if SSO suddenly cannot be used for any number of reasons (non-payment, IdP token is revoked and the person who has access to generate a new one is on vacation, etc. etc)

**Review by wschurman — APPROVED** [review]
